with-firebase-saml-login/App.js

```diff
@@ -0,0 +1,126 @@
+// Import third-party dependencies
+import React from 'react'
+import { View, Button, LogBox, StyleSheet, Text } from 'react-native'
+import * as WebBrowser from 'expo-web-browser';
+import * as Linking from 'expo-linking';
+
+import firebase from 'firebase'
+
+const firebaseConfig = {// !!!!!PUT YOUR FIREBASE CONFIG HERE!!!!!
+  apiKey: "AIzaSyAlZruO2T_JNOWn4ysfX6AryR6Dzm_VVaA",
+  authDomain: "blobtest-36ff6.firebaseapp.com"
+};
+
+if (!firebase.apps.length) {
+  firebase.initializeApp(firebaseConfig);
+}
+
+// Firebase sets some timeers for a long period, which will trigger some warnings. Let's turn that off for this example
+LogBox.ignoreLogs([`Setting a timer for a long period`]);
+/**
+ * @author Tag Howard
+ */
+export default class App extends React.Component {
+  state = {
+    redirectData: null,
+    loggedIn: false,
+    userCredential: null
+  };
+
+  render() {
+    return (
+      <View style={styles.container}>
+        {this.state.loggedIn && (
+          <>
+            <Text>
+              {"Profile object: " + this.state.userCredential.additionalUserInfo.profile}
+            </Text>
+            <Text>
+              {"Is this the first time firebase has seen this user: " + this.state.userCredential.additionalUserInfo.isNewUser}
+            </Text>
+            <Text>
+              {"Provider ID: " + this.state.userCredential.additionalUserInfo.providerId}
+            </Text>
+            <Text>
+              {"Username: " + this.state.userCredential.additionalUserInfo.username}
+            </Text>
+            <Text>
+              {"Operation type (should be 'login' or similar): " + this.state.userCredential.operationType}
+            </Text>
+            <Text>
+              {"Display name: " + this.state.userCredential.user.displayName}
+            </Text>
+            <Text>
+              {"Email: " + this.state.userCredential.user.email}
+            </Text>
+            <Text>
+              {"SSO Metadata: " + this.state.userCredential.user.metadata}
+            </Text>
+            <Text>
+              {"Provider data: " + this.state.userCredential.user.providerData}
+            </Text>
+            <Text>
+              {"UID: " + this.state.userCredential.user.uid}
+            </Text>
+          </>
+        )}
+        {!this.state.loggedIn && (
+          <>
+            <Text style={styles.header}>Redirect Example</Text>
+            <Button onPress={this._openAuthSessionAsync} title="SSO Login" />
+          </>
+        )}
+      </View>
+    );
+  }
+
+  _openAuthSessionAsync = async () => {
+    try {
+      let result = await WebBrowser.openAuthSessionAsync(
```

Comment:
Have you checked to see if auth session could be used with this type of authentication to reduce user facing complexity and add web support?

Reply:
No I hadn't, does it use https protocol rather than localhost? If so I will try rewriting the example (and my own app's authentication) that way.

Reply:
Actually scratch that, I did look into it but the documentation does not really work for SAML.

> AuthSession can be used for any OAuth or OpenID Connect provider

Reply:
Or at least the model I am working with personally